src/probnum/quad/solvers/_bayesian_quadrature.py

```diff
@@ -215,17 +225,19 @@ def from_problem(
             policy = RandomPolicy(batch_size, measure.sample)
         elif policy == "vdc":
             policy = VanDerCorputPolicy(batch_size, measure)
-        elif policy == "us_rand":
+        # all random max acquisition policies
+        elif policy in ["us_rand", "mi_rand", "ivr_rand"]:
             policy = RandomMaxAcquisitionPolicy(
                 batch_size=1,
-                acquisition_func=WeightedPredictiveVariance(),
-                n_candidates=us_rand_n_candidates,
+                acquisition_func=acquisition_dict[policy[:-5]],
```

Comment:
This seems somewhat dangerous to me (and it took me a while to understand what is going on here).

Reply:
I added  a comment such that the string slice is a bit more clear. 

The added assert makes sure that the if-statement is implemented correctly in case someone ever adds a new random policy (exception raise is not needed as this is not expected to be not true). 

Is it more readable like this?

Reply:
Thanks, that is hopefully more readable.